e2e-tests/src/test/fee.rs

```diff
@@ -0,0 +1,129 @@
+    utility::BlocksApi,
+    waiting::{AlephWaiting, BlockStatus},
+    AccountId, SignedConnection, TxStatus,
+};
+use once_cell::sync::Lazy;
+
+use crate::{config::setup_test, fee::current_fees, transfer::setup_for_transfer};
+
+/// In order to increase the block occupancy we need to transfer funds to a lot of accounts. This
+/// array contains the accounts we will be transferring funds to.
+static DESTINATIONS: Lazy<Vec<AccountId>> = Lazy::new(|| {
+    (0..1400)
+        .map(|i| keypair_from_string(&format!("//{i}")).account_id().clone())
+        .collect()
+});
+
+/// The level of occupancy in a block.
+enum BlockOccupancy {
+    Low,
+    High,
+}
+
+/// Ensures that the fee multiplier is adjusted according to the block occupancy.
+#[tokio::test]
+pub async fn fee_calculation() -> anyhow::Result<()> {
+    let config = setup_test();
+    let (connection, to) = setup_for_transfer(config).await;
+
+    let minimal_multiplier = FixedU128::from(1);
+
+    let (no_traffic_fee, no_traffic_multiplier) =
+        current_fees(&connection, to.clone(), None, 1).await;
+    assert_eq!(
+        no_traffic_multiplier, minimal_multiplier,
+        "In the beginning the fee multiplier should be equal to the minimal value",
+    );
+
+    fill_blocks(BlockOccupancy::Low, 10, &connection).await;
+
+    let (low_traffic_fee, low_traffic_multiplier) =
+        current_fees(&connection, to.clone(), None, 1).await;
+    assert_eq!(
+        low_traffic_multiplier, no_traffic_multiplier,
+        "Low traffic shouldn't affect the fee multiplier",
+    );
+    // This might fail if the incremented nonce has longer encoding. Just restart the test.
+    assert_eq!(
+        no_traffic_fee, low_traffic_fee,
+        "Low traffic shouldn't affect the fee"
+    );
+
+    fill_blocks(BlockOccupancy::High, 10, &connection).await;
+
+    let (high_traffic_fee, high_traffic_multiplier) =
+        current_fees(&connection, to.clone(), None, 1).await;
+    assert!(
+        high_traffic_multiplier > low_traffic_multiplier,
+        "High traffic should lead to higher fee multiplier",
+    );
+    assert!(
+        high_traffic_fee > low_traffic_fee,
+        "High traffic should lead to higher fee"
+    );
+
+    fill_blocks(BlockOccupancy::High, 10, &connection).await;
+
+    let (highest_traffic_fee, highest_traffic_multiplier) =
+        current_fees(&connection, to.clone(), None, 1).await;
+    assert!(
+        highest_traffic_multiplier > high_traffic_multiplier,
+        "High traffic should lead to higher fee multiplier",
+    );
+    assert!(
+        highest_traffic_fee > high_traffic_fee,
+        "High traffic should lead to higher fee"
+    );
+
+    let now = connection.get_best_block().await.unwrap().unwrap();
+    connection
+        .wait_for_block(|n| n >= now + 10, BlockStatus::Finalized)
+        .await;
+
+    let (after_traffic_fee, after_traffic_multiplier) =
+        current_fees(&connection, to.clone(), None, 1).await;
```

Comment:
I've found myself going back and forth a couple of time to check what `None` and `1` are -- it'd be a tad more readable if these two were extracted to a value (or a `const`) and reused.